### PR TITLE
Run test and lint during CI

### PR DESCRIPTION
**Pull request by lahsivjar**
Fixes lint and runs both lint and test in CI on new pushes and pull requests.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,7 +13,9 @@ fmt: tools/go.mod
 
 lint: tools/go.mod
 	go run -modfile=tools/go.mod honnef.co/go/tools/cmd/staticcheck -checks=all ./...
-	go list -m -json $(MODULE_DEPS) | go run -modfile=tools/go.mod go.elastic.co/go-licence-detector \
+	# See https://github.com/elastic/go-licence-detector/issues/22
+	# TODO: Enable after the licence-detector issue is fixed
+	# go list -m -json $(MODULE_DEPS) | go run -modfile=tools/go.mod go.elastic.co/go-licence-detector \
 		-includeIndirect -rules tools/notice/rules.json -validate
```
Comment (axw) [inline Makefile:19]:
```suggestion
		# -includeIndirect -rules tools/notice/rules.json -validate
```

**Review by axw — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (lahsivjar) [reply Makefile]: I updated the `go-license-detector` to remove `-validate` and add a `NOTICE.txt` (the template was available but not used so far it seems).

**Review by axw — APPROVED** [review]
LGTM. I don't think NOTICE.txt is required.

**Comment by lahsivjar** [comment]
Thanks, removed `NOTICE.txt`.

**Pull request merged**